src/services/audius-api-client/ResponseAdapter.ts

```diff
@@ -118,3 +127,74 @@ export const makeTrack = (track: APITrack): UserTrackMetadata | undefined => {
 
   return marshalled
 }
+
+export const makePlaylist = (
+  playlist: APIPlaylist
+): UserCollection | undefined => {
+  const decodedPlaylistId = decodeHashId(playlist.id)
+  const decodedOwnerId = decodeHashId(playlist.user_id)
+  const user = makeUser(playlist.user)
+  if (!decodedPlaylistId || !decodedOwnerId || !user) {
+    return undefined
+  }
+
+  const saves = playlist.followee_favorites
+    .map(makeFavorite)
+    .filter(removeNullable)
+
+  const reposts = playlist.followee_reposts
+    .map(makeRepost)
+    .filter(removeNullable)
+
+  const playlistContents = {
+    track_ids: playlist.added_timestamps.map(ts => ({
+      track: decodeHashId(ts.track_id),
+      time: ts.timestamp
+    }))
+  }
+
+  const tracks = playlist.tracks.map(track => makeTrack(track))
+
+  const marshalled = {
+    ...playlist,
+    user,
+    tracks,
+    playlist_id: decodedPlaylistId,
+    playlist_owner_id: decodedOwnerId,
+    followee_saves: saves,
+    followee_reposts: reposts,
+    save_count: playlist.favorite_count,
+    playlist_contents: playlistContents,
+
+    // Fields to prune
+    id: undefined,
+    user_id: undefined,
+    followee_favorites: undefined,
+    artwork: undefined,
+    favorite_count: undefined,
+    added_timestamps: undefined
+  }
+
+  delete marshalled.id
+  delete marshalled.user_id
+  delete marshalled.followee_favorites
+  delete marshalled.artwork
+  delete marshalled.favorite_count
+  delete marshalled.added_timestamps
+
+  // @ts-ignore
+  return marshalled as UserCollection
+}
+
+export const makeActivity = (
+  activity: APIActivity
+): UserTrackMetadata | UserCollection | undefined => {
+  const { item, item_type: type } = activity
+  if (type === 'track') {
+    return makeTrack(item as APITrack)
```

Comment:
Once we use a discriminated union for APIActivity, we can lose these casts

Reply:
Really annoyingly, ts can't seem to figure this out if we destructure activity, so i had to fully qualify
```
switch (activity.item_type) {
    case 'track':
      return makeTrack(activity.item)
    case 'playlist':
      return makePlaylist(activity.item)
  }
```